test/spec/Feature/Query/RpcSpec.hs

```diff
@@ -24,50 +24,65 @@ spec :: PgVersion -> SpecWith ((), Application)
 spec actualPgVersion =
   describe "remote procedure call" $ do
     context "a proc that returns a set" $ do
-      it "returns paginated results" $ do
-        request methodPost "/rpc/getitemrange"
-                (rangeHdrs (ByteRangeFromTo 0 0))  [json| { "min": 2, "max": 4 } |]
-           `shouldRespondWith` [json| [{"id":3}] |]
-            { matchStatus = 200
-            , matchHeaders = ["Content-Range" <:> "0-0/*"]
-            }
-        request methodGet "/rpc/getitemrange?min=2&max=4"
-                (rangeHdrs (ByteRangeFromTo 0 0)) ""
-           `shouldRespondWith` [json| [{"id":3}] |]
-            { matchStatus = 200
-            , matchHeaders = ["Content-Range" <:> "0-0/*"]
-            }
-        request methodHead "/rpc/getitemrange?min=2&max=4"
-            (rangeHdrs (ByteRangeFromTo 0 0)) ""
-          `shouldRespondWith`
-            ""
-            { matchStatus = 200
-            , matchHeaders = [ matchContentTypeJson
-                             , "Content-Range" <:> "0-0/*" ]
-            }
+      context "returns paginated results" $ do
+        it "using the Range header" $
+          request methodGet "/rpc/getitemrange?min=2&max=4"
+                  (rangeHdrs (ByteRangeFromTo 1 1)) mempty
+             `shouldRespondWith` [json| [{"id":4}] |]
+              { matchStatus = 200
+              , matchHeaders = ["Content-Range" <:> "1-1/*"]
+              }
 
-      it "includes total count if requested" $ do
-        request methodPost "/rpc/getitemrange"
-                (rangeHdrsWithCount (ByteRangeFromTo 0 0))
-                [json| { "min": 2, "max": 4 } |]
-           `shouldRespondWith` [json| [{"id":3}] |]
-            { matchStatus = 206 -- it now knows the response is partial
-            , matchHeaders = ["Content-Range" <:> "0-0/2"]
-            }
-        request methodGet "/rpc/getitemrange?min=2&max=4"
-                (rangeHdrsWithCount (ByteRangeFromTo 0 0)) ""
-           `shouldRespondWith` [json| [{"id":3}] |]
-            { matchStatus = 206
-            , matchHeaders = ["Content-Range" <:> "0-0/2"]
-            }
-        request methodHead "/rpc/getitemrange?min=2&max=4"
-            (rangeHdrsWithCount (ByteRangeFromTo 0 0)) ""
-          `shouldRespondWith`
-            ""
-            { matchStatus = 206
-            , matchHeaders = [ matchContentTypeJson
-                             , "Content-Range" <:> "0-0/2" ]
-            }
+        it "using limit and offset" $ do
+          post "/rpc/getitemrange?limit=1&offset=1" [json| { "min": 2, "max": 4 } |]
+             `shouldRespondWith` [json| [{"id":4}] |]
+              { matchStatus = 200
+              , matchHeaders = ["Content-Range" <:> "1-1/*"]
```

Comment:
Fortunately the spec doesn't restrict the `Content-Range` to only GET https://www.rfc-editor.org/rfc/rfc9110.html#name-content-range

So this response should be fine.